src/ui/ModelProperty.cpp

```diff
@@ -0,0 +1,41 @@
+#include "ModelProperty.hh"
+
+#include <algorithm>
+
+namespace ra {
+namespace ui {
+
+int ModelPropertyBase::s_nNextKey = 1;
+std::vector<ModelPropertyBase*> ModelPropertyBase::s_vProperties;
```

Comment:
Since the hash only contains the key, this provides a way to get from the key back to the property. Property instances are static singletons, this only holds a reference.

Reply:
Ok, got you.